pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -0,0 +1,504 @@
+    List<String> sortedColumns = indexingConfig.getSortedColumn();
+    assert sortedColumns.size() <= 1;
+    if (sortedColumns.size() == 1) {
+      _sortedColumn = sortedColumns.get(0);
+    }
+    // Check whether configs overwritten by Hadoop properties.
+    if (_properties.getProperty(SORTED_COLUMN) != null) {
+      _logger.info("Overriding sorted column to: {}", _properties.getProperty(SORTED_COLUMN));
+      _sortedColumn = _properties.getProperty(SORTED_COLUMN);
+    }
+  }
+
+  private void fetchResizingConfig() {
+    if (_properties.getProperty(MINIMUM_NUMBER_OF_OUTPUT_FILES) != null) {
+      _numberOfOutputFiles = Integer.parseInt(_properties.getProperty(MINIMUM_NUMBER_OF_OUTPUT_FILES));
+    } else {
+      _numberOfOutputFiles = 0;
+    }
+  }
+
+  /**
+   * Finds the avro file in the input folder, and returns its avro schema
+   * @param inputPathDir Path to input directory
+   * @return Input schema
+   * @throws IOException exception when accessing to IO
+   */
+  private static Schema getSchema(String inputPathDir) throws IOException {
+    FileSystem fs = FileSystem.get(new Configuration());
+    Schema avroSchema = null;
+    for (String input : inputPathDir.split("\\s*,\\s*")) {
+      Path inputPath = new Path(input);
+      for (FileStatus fileStatus : fs.listStatus(inputPath)) {
+        if (fileStatus.isFile() && fileStatus.getPath().getName().endsWith(".avro")) {
+          _logger.info("Extracting schema from " + fileStatus.getPath());
+          avroSchema = extractSchemaFromAvro(fileStatus.getPath());
+          break;
+        }
+      }
+    }
+    return avroSchema;
+  }
+
+  /**
+   * Extracts avro schema from avro file
+   * @param avroFile Input avro file
+   * @return Schema in avro file
+   * @throws IOException exception when accessing to IO
+   */
+  private static Schema extractSchemaFromAvro(Path avroFile) throws IOException {
+    DataFileStream<GenericRecord> dataStreamReader = getAvroReader(avroFile);
+    Schema avroSchema = dataStreamReader.getSchema();
+    dataStreamReader.close();
+    return avroSchema;
+  }
+
+  /**
+   * Helper method that returns avro reader for the given avro file.
+   * If file name ends in 'gz' then returns the GZIP version, otherwise gives the regular reader.
+   *
+   * @param avroFile File to read
+   * @return Avro reader for the file.
+   * @throws IOException exception when accessing to IO
+   */
+  private static DataFileStream<GenericRecord> getAvroReader(Path avroFile) throws IOException {
+    FileSystem fs = FileSystem.get(new Configuration());
+    if (avroFile.getName().endsWith("gz")) {
+      return new DataFileStream<>(new GZIPInputStream(fs.open(avroFile)), new GenericDatumReader<>());
+    } else {
+      return new DataFileStream<>(fs.open(avroFile), new GenericDatumReader<>());
+    }
+  }
+
+  private ArrayList<FileStatus> getDataFilesFromPath(FileSystem fs, Path inBaseDir) throws IOException {
```

Comment:
this code is elsewhere in open source, we should use that as there are more formats now

Reply:
Checked the code. The scenario here is different since we're only interested in some certain formats.

Reply:
what about orc?